src/features/payment/components/IframeCardForm/IframeCardForm.tsx

```diff
@@ -0,0 +1,286 @@
+import React from "react";
+import { Box } from "@mui/material";
+import { useTranslation } from "react-i18next";
+import {
+  getSessionItem,
+  SessionItems,
+} from "../../../../utils/storage/sessionStorage";
+import { FormButtons } from "../../../../components/FormButtons/FormButtons";
+import { PaymentMethod } from "../../../../features/payment/models/paymentModel";
+import { Field, RenderField } from "./IframeCardField";
+
+interface Props {
+  loading?: boolean;
+  onCancel: () => void;
+  onSubmit?: (bin: string) => void;
+  hideCancel?: boolean;
+}
+
+interface FieldFormStatus {
+  isValid: boolean;
+  errorCode: null | string;
+  errorMessage: null | string;
+}
+
+interface BuildResp {
+  sessionId: string;
+  securityToken: string;
+  fields: Array<Field>;
+}
+
+export enum IdFields {
+  "CARD_NUMBER" = "CARD_NUMBER",
+  "EXPIRATION_DATE" = "EXPIRATION_DATE",
+  "SECURITY_CODE" = "SECURITY_CODE",
+  "CARDHOLDER_NAME" = "CARDHOLDER_NAME",
+}
+type FieldsFormStatus = Map<keyof typeof IdFields | string, FieldFormStatus>;
+
+const initialFormStatus: FieldFormStatus = {
+  isValid: false,
+  errorCode: null,
+  errorMessage: null,
+};
+
+const fieldformStatus: FieldsFormStatus = new Map();
+Object.values(IdFields).forEach((k) => {
+  fieldformStatus.set(k as keyof typeof IdFields, initialFormStatus);
+});
+
+// eslint-disable-next-line sonarjs/cognitive-complexity
+export default function IframeCardForm(props: Props) {
+  const { loading = true, onCancel, onSubmit = () => null, hideCancel } = props;
+  const [error, setError] = React.useState(false);
+  const [form, setForm] = React.useState<BuildResp>();
+  const [spinner, setSpinner] = React.useState(loading);
+  const [enabledForm, setEnabledForm] = React.useState(false);
+  const [_sdkBuildIstance, setSdkBuildIstance] = React.useState({});
+  // this dummy state is only used to permorm a component udpate, not the best solution but works
+  const [, setDummyState] = React.useState(0);
+
+  const { hostname, protocol, port } = window.location;
+
+  const calculateFormValidStatus = (
+    fieldformStatus: Map<string, FieldFormStatus>
+  ) => Array.from(fieldformStatus.values()).every((el) => el.isValid);
+
+  React.useEffect(() => {
+    if (!form) {
+      const fetchForm = async () => {
+        try {
+          const { paymentMethodId } = getSessionItem(
+            SessionItems.paymentMethod
+          ) as PaymentMethod;
+          const response = await fetch(
+            `/ecommerce/checkout/v1/payment-methods/${paymentMethodId}/preauthorization`,
```

Comment:
👉🏻 Change this from preauthorization to sessions

Reply:
ping :-)